src/imitation/algorithms/dagger.py

```diff
@@ -6,39 +6,76 @@
 policy.
 """
 
+import abc
 import dataclasses
+import logging
 import os
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
 
-import cloudpickle
 import gym
 import numpy as np
-import tensorflow as tf
-from stable_baselines.common.policies import BasePolicy
+import torch as th
+from stable_baselines3.common import utils
 
-from imitation.algorithms.bc import BC, set_tf_vars
+from imitation.algorithms import bc
 from imitation.data import rollout, types
 from imitation.util import util
 
 
-def linear_beta_schedule(rampdown_rounds: int) -> Callable[[int], float]:
-    """Linearly-decreasing schedule for beta (% of time that user act is used).
+class BetaSchedule(abc.ABC):
+    """
+    Determines the value of beta (% of time that demonstrator action is used) over the
+    progression of training rounds.
+    """
 
-    Args:
-      rampdown_rounds: number of rounds over which to anneal beta. Rounds
-        are assumed to be numbered 0, 1, 2, ….
+    @abc.abstractmethod
+    def __call__(self, round_num: int) -> float:
+        """Gives the value of beta for the current round.
 
-    Returns:
-      schedule: function that takes current round number (0, or 1, or 2, etc.)
-        as input & returns current beta as float.
+        Args:
+            round: the current round number. Rounds are assumed to be numbered 0, 1, 2,
+              etc.
+
+        Returns:
+            beta: the fraction of the time to sample a demonstrator action. Robot
+              actions will be sampled the remainder of the time.
+        """
+
+
+class LinearBetaSchedule(BetaSchedule):
```

Comment:
The reason I refactored these from functions to classes is in order to use `torch.save` and `torch.load`. These save and load methods can't handle functions (I think because they use pickling under the hood), but they can handle classes.

Reply:
You could use `torch.save` with `pickle_module=cloudpickle` which can handle functions. I'm not sure this is worth it though if this is the only issue with straightforward `pickle`. Thoughts @qxcv?

Reply:
I'm happy with either solution. I don't see a pressing reason to switch now that Scott has made this work, but if we run into the same problem again then we can refactor our saves to always go through cloudpickle.